pkg/kube/log_tail.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"bufio"
+	"bytes"
+	"strings"
+)
+
+// LogTail interface allows to store last N lines of log written to it
+type LogTail interface {
+	Write(p []byte) (int, error) // Write log line(s) to circular buffer
+	ToString() string            // Join stored log lines to one string separated by newline
```

Comment:
Can we have the comments above the functions so that they build up a proper godoc format.